example/spec/parsePdr/ParsePdrSuccessSpec.js

```diff
@@ -169,4 +170,59 @@ describe('Parse PDR workflow', () => {
       expect(record.status).toEqual('completed');
     });
   });
+
+  describe('When a workflow is configured to make a choice based on the output of a Cumulus task', () => {
+    let executionStatus;
+
+    beforeAll(async () => {
+      const executionArn = workflowExecution.executionArn;
+      executionStatus = await apiTestUtils.getExecutionStatus({
+        prefix: config.stackName,
+        arn: executionArn
+      });
+    });
+
+    it('branches according to the CMA output', async () => {
+      expect(executionStatus.executionHistory).toBeTruthy();
+      const events = executionStatus.executionHistory.events;
+
+      // the output of the CheckStatus is used to determine the task of choice
+      const checkStatus = 'CheckStatus';
+      const stopStatus = 'StopStatus';
+      const pdrStatusReport = 'PdrStatusReport';
+
+      let choiceVerified = false;
+      for (let i = 0; i < events.length; i += 1) {
+        const currentEvent = events[i];
+        if (currentEvent.type === 'TaskStateExited' &&
+        get(currentEvent, 'stateExitedEventDetails.name') === checkStatus) {
+          const output = JSON.parse(get(currentEvent, 'stateExitedEventDetails.output'));
+          const isFinished = output.payload.isFinished;
+
+          // get the next task executed
+          let nextTask;
+          while (!nextTask && i < events.length - 1) {
```

Comment:
Same as my comment on the iterator above

Reply:
In the ParsePdr workflow, after CheckStatus state, before the next task, there is CheckAgainChoice state which has type 'ChoiceStateEntered' and 'ChoiceStateExited', so the events[i+2] will not work now.

Reply:
Ah, that makes sense.  thanks for the explanation 👍